### PR TITLE
fix(profile): replace adding http(s) to urls with elgg_normalize_url

### LINKED ISSUE #6569: Use elgg_normalize_url() in profile edit action
Right now we're doing this:

```
    if ($value && $valuetype == 'url' && !preg_match('~^https?\://~i', $value)) {
        $value = "http://$value";
    }
```

Is there a reason we're not using elgg_normalize_url()? Quick search didn't pull up anything. There are use cases for non-http(s) URLs in this field.

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #6569

### CODE REVIEW

File: actions/profile/edit.php [hunk 1]
```diff
@@ -51,8 +51,8 @@ function profile_array_decoder(&$v) {
 		forward(REFERER);
 	}
 
-	if ($value && $valuetype == 'url' && !preg_match('~^https?\://~i', $value)) {
-		$value = "http://$value";
+	if ($value && $valuetype == 'url') {
+		$value = elgg_normalize_url($value);
```
Comment (mrclay) [inline actions/profile/edit.php]: We're sure "google.com" will become "http://google.com" ?

Comment (mrclay) [inline actions/profile/edit.php]: I know elgg_normalize_url() got some improvements, so maybe so.

Comment (mrclay) [inline actions/profile/edit.php]: I don't think we want this. elgg_normalize_url() will passthrough URLs like "javascript:alert(1)".

Comment (jdalsem) [inline actions/profile/edit.php]: hmmm there's a point, maybe we can make an `elgg_normalize_url($value, $strip_unsafe_true)`

**Comment by jdalsem** [comment]
before the fix `ftp://test.com` would become `http://ftp://test.com`

**Comment by mrclay** [comment]
I'm OK with users entering silly URLs.

**Comment by jdalsem** [comment]
> I'm OK with users entering silly URLs.

is ftp://test.com such a silly one? it is allowed by html5 as a valid URL

**Comment by mrclay** [comment]
The question is what protocols Elgg wants to support out of the box.

**Comment by mrclay** [comment]
I would suggest a hook-filterable whitelist of http and https.

**Pull request closed without merging**